doc/user/tutorials/making_your_first_app.rst

```diff
@@ -0,0 +1,510 @@
+  put it there.
+
+Registering the app with ARMI
+=============================
+The ARMI Framework contains features to run the "main loop" of typical applications. In
+order to get access to these, we must register our new app with the ARMI framework. To do
+this, we put the following code in the top-level :file:`__init__.py` module:
+
+.. code-block:: python
+    :caption: ``myapp/__init__.py``
+
+    import armi
+    from myapp import app
+    armi.configure(app.ExampleApp())
+
+Defining the app class
+======================
+We define our app in the :file:`myapp/app.py` module. For this example, the app class is
+relatively simple: it will just register our one custom plugin. We will actually create
+the plugin shortly.
+
+.. admonition:: Apps vs. plugins vs. interfaces
+
+    ARMI-based methodologies are broken down into three layers of abstraction. Apps are
+    collections of plugins intended to perform analysis on a certain type of reactor.
+    Plugins are independent and mixable collections of relatively arbitrary code that
+    might bring in special materials, contain certain engineering methodologies, and/or
+    Interfaces with one or more physics kernels.
+
+.. code-block:: python
+    :caption: ``myapp/app.py``
+
+    import armi
+    from armi.apps import App
+
+    from myapp.plugin import DummyPhysicsPlugin
+
+    class ExampleApp(App):
+        def __init__(self):
+            # activate all built-in plugins
+            App.__init__(self)
+
+            # register our plugin with the plugin manager
+            self._pm.register(DummyPhysicsPlugin) 
+
+        @property
+        def splashText(self):
+            return "** My Example App **"
+
+
+Defining the physics plugin
+===========================
+Now we will create the plugin that will coordinate our dummy physics modules. 
+
+.. admonition:: What are plugins again? 
+
+    Plugins are the basic modular building block of ARMI-based apps. In some cases, one plugin
+    will be associated with one physics kernel (like MCNP). This is a reasonable practice when
+    you expect to be mixing and matching various combinations of plugins between related
+    teams. It is also possible to have a plugin that performs a whole cacophony of analyses
+    using multiple codes, which some smaller research teams may find preferable. The
+    flexibility is very broad.
+
+    See :py:mod:`armi.plugins` more for info.
+
+Plugin code can exist in any directory structure in an app. In this very simple app we
+simply put it in the :file:`myapp/plugins.py` file.
+
+.. note:: For "serious" plugins, we recommend mirroring the ``armi/physics/[subphysics]``
+    structure of the ARMI framework physics plugin folder.
+
+We will start the plugin simply by pointing to the two physics kernels we wish to
+register. We hook them in and tell ARMI the ``ORDER`` they should be run in based on
+the built-in ``STACK_ORDER`` attribute. We will come back to this plugin definition later
```

Comment:
`STACK_ORDER` has always been super confusing to me, so this may warrant some additional documentation or some discussion that this links to.

Reply:
added a hyperlink to the docstring of it